src/domexpression.js

```diff
@@ -159,6 +159,21 @@ var _ = Mavo.DOMExpression = $.Class({
 					return "";
 				}
 
+				if (env.value instanceof Promise) {
+					env.value
+						.then(value => {
+							env.expr.function = () => value;
```

Comment:
Why are we updating the function? It seems to me that it's the return value that we need to handle specially, and leave the function as-is, but since you've worked on it you may have better insight here.

Reply:
My insights are the following. Please, correct me. I might be wrong.

1. On every expression update, its value (`env.value`) is a result returned by the `env.expr.function` function.
2. This function is compiled based on a parsed expression when the expression is being evaluated for the first time and is never recompiled in the future. So, if this function returns a promise, it will always be returning a promise, not its resolved or rejected value.
3. We could change this function by constructing a new expression based on data returned by a resolved or rejected promise, parse it and let Mavo compile the corresponding function for its evaluation. But the data could be anything. How can we build an expression based on an object, for example? I suppose we can't.
4. So, the only way I could find was just updating the function that computes the expression's value since its sole purpose is to return the resolved value.

Reply:
Indeed, but overwriting `expression.function`, while it may work, is a bit of a hack. It defeats the purpose of having a parsed function, since it needs to be re-evaluated every time.
The solution I'd go with is to check if we have a promise in DOMExpression#update(), and run the rest of its code asynchronously in those cases once the promise is fulfilled.

Reply:
Thank you very much for your patience! I think I finally got your idea. I refactored the code. I still have a couple of doubts. I'll share them with you in the comments below.